client/src/app/site/deployment-center/deployment-center-setup/wizard-logic/deployment-center-state-manager.ts

```diff
@@ -463,13 +463,32 @@ export class DeploymentCenterStateManager implements OnDestroy {
     this._localStorageService.setItem(storedPermissionItem.id, storedPermissionItem);
   }
 
-  public getVstsPassthroughHeaders(appendMsaPassthroughHeader: boolean = false): Headers {
-    const headers = new Headers();
-    headers.append('Content-Type', 'application/json');
-    headers.append('Accept', 'application/json');
-    headers.append('Vstsauthorization', `Bearer ${this._vstsApiToken}`);
-    headers.append('MsaPassthrough', `${appendMsaPassthroughHeader}`);
-    return headers;
+  private _getVstsToken(): Observable<{ status: string; statusMessage: string; result: any }> {
+    return this._portalService
+      .getAdToken('azureTfsApi')
+      .first()
+      .switchMap(tokenData => {
+        if (tokenData == null || tokenData.result == null || tokenData.result.token == null || !parseToken(tokenData.result.token)) {
```

Comment:
Should you instead use !tokenData !tokenData.result.token . Tha will cover more cases like undefined empty string etc

Reply:
[getAdToken()](https://github.com/vinodkumar3/azure-functions-ux/blob/60c4c1d47091953b13a1493284fb53fd5f99ad4d/client/src/app/shared/services/portal.service.ts#L261) returns null if it is not able to successfully get token, hence I only checked for null.
But I guess no harm in checking undefined. Will update.